### PR TITLE
Simple Sidebar Implementation - rs2443-kj333

### PR DESCRIPTION
**Pull request by sedhainr2004**
## Overview

This pull request implements the sidebar navigation for the home page of the client webpage for IMA. 

## Replication

```
cd client
npm install
npm start
```

## Changes Made

`client/`

- Installed MUI material icons via npm 

`components/`

- Added Sidebar.css, Sidebar.js

`Sidebar.css`

-  Implemented the basic design for the sidebar following the "search/results high-fi" design in Figma. 

`Sidebar.js`

- Implemented the sidebar container and created 4 categories: roles, interests, graduation year, and location following the .component class in Sidebar.css. NOTE: There are no event listeners in the component class for the moment.

## Test Coverage

We manually tested the sidebar by running the application on localhost via npm start. We resized the windows to make sure the sidebar wasn't acting weird with different windows. 

## Next Steps 

- Now we're using the useState hooks to add states to each of the components to create a toggle function where the values are true and false for each of the components which determines the dropdown menu visibility
- The dropdown menu will consist of checkboxes that correspond to popular sub-categories within the category (e.g roles will have a dropdown checkbox of developer, designer, PM, and other)
- Store the checked subcategories in an array so those values can be passed to the App.js or whatever file handles the visibility of profiles so it's easier when we implement the filter function.

## Screenshots

<details>

  <summary>Sidebar</summary>

Screen Shot 2024-03-21 at 9 43 23 PM

  

</details>

### CODE REVIEW

**Review by phoebe-qian — COMMENTED** [review]
Good job on your code and I like how you guys were able to stick with the figma design to make everything look as standardized as possible

File: client/src/components/Sidebar.js
Comment (Akul9092) [inline client/src/components/Sidebar.js:1]: From this file, I really liked how you kept the categories, ie, roles, interests, grad years and location, as arrays, which allows developers in the future to easily alter these options in the future when required. Moreover, I also liked how you guys implemented ‘useState’ which allows for dynamic changes on the webpage when users check/uncheck dropdowns. 

One improvement you guys could implement is possibly the formatting of the code, like indentation etc. At first glance, I found the code slightly difficult to follow given all the line breaks and 3 div classes on the file.

File: client/src/components/Sidebar.css
Comment (Akul9092) [inline client/src/components/Sidebar.css:1]: This file demonstrates an impressive attention to detail in its styling,  alignment, structure, font size, and family. The inclusion of comments in the CSS file enhances its clarity, enabling future developers to easily get up to pace.

An enhancement I propose involves integrating variables. Incorporating variables for recurring colors and fonts streamlines the styling process and facilitates future modifications. For example:

:root {
  --websiteTheme: #1e90ff;
  --outlineColor: #ffffff;
}
By employing variables like these, maintaining consistency and flexibility in the stylesheet becomes notably more manageable.

**Review by Akul9092 — COMMENTED** [review]
I'm impressed with the additions you've made to the Sidebar as they're quite effective. I noticed you've incorporated some techniques such as useState and comments, which I intend to adopt in my future code. It's particularly commendable how the styling in your CSS file aligns seamlessly with the Figma designs, promoting a cohesive visual experience. Keep up the great work!

**Pull request merged**